language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/contexts/AbstractDocumentServiceContext.java

```diff
@@ -123,6 +126,15 @@ public WorkspaceManager workspace() {
         return this.currentDocImports;
     }
 
+    @Override
+    public Optional<Module> currentModule() {
+        if (this.currentModule == null) {
+            this.currentModule = this.workspaceManager.module(this.filePath).orElse(null);
```

Comment:
Let's keep the currentDocument as  the lazy loading field. Then we can  calculate module, package, project on top of it. We can push it on post SLP6.

Reply:
+1